craft_application/commands/test.py

```diff
@@ -34,14 +37,18 @@ class TestCommand(base.AppCommand):
     help_msg = "Run project tests"
     overview = dedent(
         """
-        Long description.
+        Execute project tests using spread.
         """
     )
     common = True
 
     def run(self, parsed_args: argparse.Namespace) -> None:  # noqa: ARG002
         """Run the command."""
         craft_cli.emit.progress("Testing project.")
-        self._services.testing.process_spread_yaml()
-        # TODO: execute spread test
-        # craft_cli.emit.message("Successfully tested project.")  # noqa: ERA001
+
+        dest = pathlib.Path(TEMP_FILE_NAME)
+        try:
+            self._services.testing.process_spread_yaml(dest)
+            self._services.testing.run_spread(dest)
+        finally:
+            dest.unlink()
```

Comment:
Does it make sense to use `tempfile.NamedTemporaryFile` here ? We could use the `TEMP_FILE_NAME` as a prefix and file would never be ovewritten

Reply:
Honestly I think I prefer having it be `.craft-spread.yaml`, since it makes it very clear what it is and if it already exists we can simply overwrite it.